pkg/spanconfig/spanconfigreconciler/reconciler.go

```diff
@@ -403,30 +415,41 @@ func (r *incrementalReconciler) reconcile(
 				}
 			}
 
-			// TODO(irfansharif): Would it be easier to just have the translator
-			// return the set of missing table IDs? We're using two transactions
-			// here, somewhat wastefully. An alternative would be to have a
-			// txn-scoped translator.
-
-			missingTableIDs, err := r.filterForMissingTableIDs(ctx, sqlUpdates)
-			if err != nil {
-				return err
-			}
+			var missingTableIDs []descpb.ID
+			var missingProtectedTimestampTargets []spanconfig.SystemTarget
+			var records []spanconfig.Record
+
+			if err := sql.DescsTxn(ctx, r.execCfg,
+				func(ctx context.Context, txn *kv.Txn, descsCol *descs.Collection) error {
+					// We're in a retryable closure, so clear any records state from
+					// previous attempts.
+					missingTableIDs = missingTableIDs[:0]
+					missingProtectedTimestampTargets = missingProtectedTimestampTargets[:0]
+					records = records[:0]
+
+					var err error
+					missingTableIDs, err = r.filterForMissingTableIDs(ctx, txn, descsCol, sqlUpdates)
+					if err != nil {
+						return err
+					}
 
-			missingProtectedTimestampTargets, err := r.filterForMissingProtectedTimestampSystemTargets(
-				ctx, sqlUpdates)
-			if err != nil {
-				return err
-			}
+					missingProtectedTimestampTargets, err = r.filterForMissingProtectedTimestampSystemTargets(
```

Comment:
Ditto re: TODO, the translator could adopt this responsibility easily instead of having the reconciler house this logic. This is easy now that the translator's txn scoped.